### PR TITLE
Added support for multiple streamdecks & config file flag

### PR DESCRIPTION
**Pull request by The-Jonsey**
- Added `-config` flag to specify a path to a config file on launch
- Added support for multiple streamdecks

### CODE REVIEW

**Review by andydotxyz — CHANGES_REQUESTED** [review]

File: main.go [hunk 1]
```diff
@@ -21,84 +21,143 @@ import (
...
+				sDInfo = append(sDInfo, api.StreamDeckInfo{
+					Cols:     int(dev.Deck.Columns),
+					Rows:     int(dev.Deck.Rows),
+					IconSize: int(dev.Deck.Pixels),
+					Page:     0,
+					Serial:   dev.Deck.Serial,
+				})
 			}
-			Listen()
+			go Listen(dev)
```
[the remark this answers is not in the document]
Comment (andydotxyz) [reply main.go:81]: If this is desirable we at least should have a time delay so it's not spinning a CPU

[the remark this answers is not in the document]
Comment (The-Jonsey) [reply main.go:82]: Added a time delay to reduce CPU usage, as it is desired for streamdeckd to always be listening for new devices

File: interface.go [hunk 2]
```diff
@@ -148,6 +162,12 @@ func HandleInput(key *api.Key, page int) {
 		runCommand("xdg-open " + key.Url)
 	}
 	if key.KeyHandler != "" {
+		var deckInfo api.StreamDeckInfo
+		for i := range sDInfo {
+			if sDInfo[i].Serial == dev.Deck.Serial {
+				deckInfo = sDInfo[i]
```
Comment (andydotxyz) [inline interface.go:168]: In other places you exit with error if there is no deck found. Will just going ahead here not cause a crash?

**Review by andydotxyz — COMMENTED** [review]
Nice, works for me - just a comment inline.
One other thought is that my old config didn't work (previewed OK but buttons didn't action) - not sure if that is a problem or not, I reset and it worked fine.

**Pull request merged**